tsdb/engine/tsm1/file_store.go

```diff
@@ -1118,6 +1130,11 @@ func (p *purger) purge() {
 			p.mu.Lock()
 			for k, v := range p.files {
 				if !v.InUse() {
+					// Remove any mmap references held by the index.
+					if p.fileStore.dereferencer != nil {
+						v.deref(p.fileStore.dereferencer)
```

Comment:
This also needs to be called within `FileStore.Replace` and `FileStore.Remove`. The `Purger` just handles removing files that are still in use by a query after a compaction has run.